src/spicy/spicyz/driver.cc

```diff
@@ -37,12 +35,15 @@ struct VisitorTypes : public spicy::visitor::PreOrder {
         : driver(driver), glue(glue), is_resolved(is_resolved) {}
 
     void operator()(hilti::declaration::Module* n) final {
-        if ( n->uid().in_memory ) {
-            // Ignore modules built by us in memory.
+        // Ignore modules built by us in memory, or builtin modules which
+        // never contain implementations of hooks for user types.
+        if ( n->uid().in_memory ||
+             (module == "hilti" || module == "spicy" || module == "spicy_rt" || module == "zeek_rt") ) {
```

Comment:
Thanks for `skipImplementation`, I missed that.

The location where I initially put this check is incorrect (even for `module` which would have been set in the previous invocation) since we still need to at least set up `module` and `path` for them (otherwise we'd e.g., fail to export builtin types). I'll fix this in an update.